src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
+
+#include <algorithm>
+
+//--------------------------------------------------------------------------------------------------
+// Container-wide versions of std algoithms.
+//--------------------------------------------------------------------------------------------------
+
+template <typename Container, typename Predicate>
+inline bool any_of(Container&& c, Predicate p) {
+    using std::begin; //ADL
```

Comment:
Activities of Daily Living?  Anti-Defamation League?

*sure, KA101 knows what ADL means!*